common/priorityQueue.py

```diff
@@ -0,0 +1,81 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
```

Comment:
I did originally implement it with heapq but we might have to update something that is currently in the heap and heapq doesn't support that.